custom_model_runner/datarobot_drum/drum/gpu_predictors/nemo_predictor.py

```diff
@@ -43,7 +44,9 @@ def model_name(self):
     def download_and_serve_model(self, openai_process: DrumServerProcess):
         if self.python_model_adapter.has_custom_hook(CustomHooks.LOAD_MODEL):
             try:
+                self.status_reporter.report_deployment("Model artifacts download started...")
```

Comment:
This doesn't really work with your static nature of deployment stages (e.g. it seems that having a load model hook is optional but you are hard-coding 4 stages)

Reply:
That "stage"  is used only for title, we can drop it at all. But if model hook will not load model, the message DRUM sends will be
```
Deployment stage 1 out of 4
NeMo Inference Server is launching...
```
and then there will be a health event about failed deployment.